gst/datarepo/gstdatareposink.c

```diff
@@ -293,6 +299,45 @@ gst_data_repo_sink_write_others (GstDataRepoSink * sink, GstBuffer * buffer)
   return GST_FLOW_OK;
 }
 
+/**
+ * @brief Function to validate flexible tensors
+ */
+static gboolean
+gst_data_repo_sink_validate_flexible_tensors (GstBuffer * buffer)
+{
+  guint num_tensors, i;
+  gsize hsize = 0;
+  GstMemory *in_mem[NNS_TENSOR_SIZE_LIMIT];
+  GstMapInfo in_info[NNS_TENSOR_SIZE_LIMIT];
+  GstTensorMetaInfo in_meta[NNS_TENSOR_SIZE_LIMIT];
+
+  g_return_val_if_fail (buffer != NULL, FALSE);
+
+  num_tensors = gst_buffer_n_tensor (buffer);
+
+  for (i = 0; i < num_tensors; i++) {
+    in_mem[i] = gst_tensor_buffer_get_nth_memory (buffer, i);
+    if (!gst_memory_map (in_mem[i], &in_info[i], GST_MAP_READ)) {
+      GST_ERROR ("Failed to fill memory info");
+      return FALSE;
+    }
+    gst_tensor_meta_info_parse_header (&in_meta[i], in_info[i].data);
+    hsize = gst_tensor_meta_info_get_header_size (&in_meta[i]);
+    gst_memory_unmap (in_mem[i], &in_info[i]);
+    gst_memory_unref (in_mem[i]);
```

Comment:
Is the ref_count of ```in_mem[i]``` increased somewhere?

Reply:
Thank you. I will remove this line